wgpu-core/src/command/mod.rs

```diff
@@ -726,6 +763,20 @@ impl<C: Clone> BasePass<C> {
 }
 
 /// Errors related to the state of a command or pass encoder.
+///
+/// `EncoderStateError::Ended` is returned immediately when an attempt is made
+/// to encode a command using an encoder that has already finished.
+///
+/// `EncoderStateError::Locked` is returned by a subsequent call to
+/// `encoder.finish()`, if there was an attempt to open a second pass on the
+/// encoder while a previous pass was still open.
+///
+/// `EncoderStateError::Invalid` is used internally by wgpu. It should usually
+/// not be seen by users of the API, since an effort should be made to provide
+/// the caller with a more specific reason for the encoder being invalid.
```

Comment:
nitpick: These seem more appropriate to put on the individual variants they describe. Did you maybe have a reason to put them up here? 🤔

Reply:
No good reason. I had noted a TODO to update the docs and then wrote the docs where the TODO was. Will update.